src/targets/github.ts

```diff
@@ -187,56 +130,25 @@ export class GithubTarget extends BaseTarget {
     const isPreview =
       this.githubConfig.previewReleases && isPreviewRelease(version);
 
-    try {
-      const response = await this.github.repos.getReleaseByTag({
-        owner: this.githubConfig.owner,
-        repo: this.githubConfig.repo,
-        tag,
-      });
-      this.logger.warn(`Found the existing release for tag "${tag}"`);
-      return response.data;
-    } catch (e: any) {
-      if (e.status !== 404) {
-        throw e;
-      }
-      this.logger.debug(`Release for tag "${tag}" not found.`);
-    }
-
-    const createReleaseParams = {
-      draft: false,
-      name: tag,
-      owner: this.githubConfig.owner,
-      prerelease: isPreview,
-      repo: this.githubConfig.repo,
-      tag_name: tag,
-      target_commitish: revision,
-      ...changes,
-    };
-
-    this.logger.debug(`Annotated tag: ${this.githubConfig.annotatedTag}`);
-    if (!isDryRun()) {
-      if (this.githubConfig.annotatedTag) {
-        await this.createAnnotatedTag(version, revision, tag);
-        // We've just created the tag, so "target_commitish" will not be used.
-        createReleaseParams.target_commitish = '';
-      }
-
-      this.logger.info(
-        `Creating a new ${
-          isPreview ? '*preview* ' : ''
-        }release for tag "${tag}"`
-      );
-      const created = await this.github.repos.createRelease(
-        createReleaseParams
-      );
-      return created.data;
-    } else {
+    if (isDryRun()) {
       this.logger.info(`[dry-run] Not creating the release`);
       return {
         id: 0,
         tag_name: tag,
         upload_url: '',
       };
+    } else {
+      const created = await this.github.repos.createRelease({
+        draft: true,
+        name: tag,
+        owner: this.githubConfig.owner,
+        prerelease: isPreview,
+        repo: this.githubConfig.repo,
+        tag_name: tag,
+        target_commitish: revision,
+        ...changes,
+      });
+      return created.data;
     }
```

Comment:
```suggestion

    const created = await this.github.repos.createRelease({
      draft: true,
      name: tag,
      owner: this.githubConfig.owner,
      prerelease: isPreview,
      repo: this.githubConfig.repo,
      tag_name: tag,
      target_commitish: revision,
      ...changes,
    });
    return created.data;
```

Reply:
Nitpick: I always prefer early returns for non-happy-path branches, where `else` is redundant. But it's just a visual difference, so I don't mind.

Reply:
Done in d7886d943388383dddec2db271e371d29488d4f8.